backend/src/src-licenses/src/main/java/org/eclipse/sw360/licenses/tools/SpdxConnector.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2016.
+ * Part of the SW360 Portal Project.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.sw360.licenses.tools;
+import org.apache.log4j.Logger;
+import org.eclipse.sw360.datahandler.thrift.SW360Exception;
+import org.eclipse.sw360.datahandler.thrift.licenses.License;
+import org.spdx.compare.LicenseCompareHelper;
+import org.spdx.compare.SpdxCompareException;
+import org.spdx.rdfparser.InvalidSPDXAnalysisException;
+import org.spdx.rdfparser.license.LicenseInfoFactory;
+import org.spdx.rdfparser.license.SpdxListedLicense;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class SpdxConnector {
+
+    private static final Logger log = Logger.getLogger(SpdxConnector.class);
+
+    public static List<String> getAllSpdxLicenseIds() {
+        return Arrays.asList(LicenseInfoFactory.getSpdxListedLicenseIds());
+    }
+
+    protected static Optional<SpdxListedLicense> getSpdxLicense(String licenseId) {
+        final SpdxListedLicense listedLicenseById;
+        try {
+            listedLicenseById = LicenseInfoFactory.getListedLicenseById(licenseId);
+        } catch (InvalidSPDXAnalysisException e) {
+            log.warn("Failed to find SpdxListedLicense with id: " + licenseId);
+            return Optional.empty();
+        }
+        return Optional.of(listedLicenseById);
+    }
+
+    public static Optional<License> getSpdxLicenseAsSW360License(String licenseId){
+        return getSpdxLicense(licenseId)
+                .flatMap(SpdxConnector::getSpdxLicenseAsSW360License);
+    }
+
+    public static Optional<License> getSpdxLicenseAsSW360License(SpdxListedLicense spdxListedLicense){
+        License license = new License()
+                .setId(spdxListedLicense.getLicenseId())
+                .setShortname(spdxListedLicense.getLicenseId())
+                .setFullname(spdxListedLicense.getName())
+                .setText(spdxListedLicense.getLicenseText())
+                .setExternalLicenseLink("https://spdx.org/licenses/" + spdxListedLicense.getLicenseId()+ ".html");
+        return Optional.of(license);
+    }
+
+    /**
+     * checks whether the given license matches the corresponding SPDX license
+     * @param license
+     * @return
+     */
+    public static boolean matchesSpdxLicenseText(License license) {
+        return matchesSpdxLicenseText(license, license.getShortname());
+    }
+
+    public static boolean matchesSpdxLicenseText(License license, String spdxId) {
+        final Optional<SpdxListedLicense> spdxLicense = getSpdxLicense(spdxId);
+        if(!spdxLicense.isPresent()) {
+            return true;
```

Comment:
Okay, I got it. Could you, perhaps, add some code comment to clarify this point inline? Just to accommodate my short memory span.